migration/migration.go

```diff
@@ -620,23 +621,24 @@ func createOrUpdateSpace(ctx context.Context, spaceRepo *space.GormRepository, i
 }
 
 func createSpace(ctx context.Context, spaceRepo *space.GormRepository, id uuid.UUID, description string) error {
-	_, err := spaceRepo.Load(ctx, id)
-	cause := errs.Cause(err)
-	newSpace := &space.Space{
-		Description: description,
-		Name:        "system.space",
-		ID:          id,
-	}
-	switch cause.(type) {
-	case errors.NotFoundError:
+	err := spaceRepo.CheckExists(ctx, id.String())
+	switch reflect.TypeOf(err) {
+	case reflect.TypeOf(&goa.ErrorResponse{}):
```

Comment:
Agree.

What about returning `errors.NewNotFoundError(tableName, id).Error()` and https://github.com/fabric8-services/fabric8-wit/blob/master/errors/errors.go#L30 instead of instead of what we return now https://github.com/hectorj2f/fabric8-wit/blob/595bd5b655b0d865d2b4455a00b925d1d1096e7b/application/repository/exister.go#L34

We return our own error objects which could be wrapped by the goa in the respective layer